src/components/Header/NavComponents/userprofile.tsx

```diff
@@ -30,6 +68,38 @@ export const UserProfile: React.FC<{}> = () => {
   const { data: agaveBalance } = useAmountAvailableToStake(address);
   const userBal = agaveBalance ? bigNumberToString(agaveBalance, 3) : "0";
 
+  const chainAddresses = useChainAddresses();
+  const { error } = useWeb3React();
+
+  // Buttons to change to every available chain
+  let popoverData: any[] = [];
```

Comment:
- This should be `const`, as you do not rebind it later, you just mutate its contents.
- Don't use `any` this openly, make an interface if one is needed, or use the correct type for the variable. Even better, you could just *not* annotate the type, and TypeScript will figure it out for you here.
- This whole `Object.entries ... forEach` section should be in a `React.useMemo` or you'll cause a rerender for every child element on every render of this component. See the React Hooks documentation for details.
- Instead of mutating an array in a `forEach`, you should `.map` on the `.entries` in order to produce a new array containing one element per input without mutation.

Reply:
I used this:
```
const popoverData = useMemo(
  () =>
    Object.entries(internalAddressesPerNetworkId).map(([id, chain]) => {
      return (
        <Button
          bg={mode({ base: "primary.500", md: "primary.500" }, "primary.500")}
          colorScheme="teal"
          size="xl"
          h="40px"
          onClick={() => changeChain(chain.chainName)}
        >
          {chain.chainName}
        </Button>
      );
    }),
  []
);
```

Reply:
This one is nearly perfect; the only remaining possibility is basically replacing ChakraUI's generic `Button` type with one that includes theming based on intent, but that's definitely outside the scope of this PR. Resolved.